app/src/main/java/io/left/meshim/activities/MainActivity.java

```diff
@@ -223,34 +237,44 @@ private void setupSettingTab() {
     private void alertDialog() {
         final  AlertDialog levelDialog;
         AlertDialog.Builder builder = new AlertDialog.Builder(this);
-        builder.setTitle("Enter username");
+        builder.setTitle(R.string.title_onboarding_username_activity);
 
         final EditText input = new EditText(MainActivity.this);
         LinearLayout.LayoutParams lp = new LinearLayout.LayoutParams(
                 LinearLayout.LayoutParams.MATCH_PARENT,
                 LinearLayout.LayoutParams.MATCH_PARENT);
         input.setLayoutParams(lp);
         builder.setView(input);
-
-        User user = User.fromDisk(this);
-        builder.setPositiveButton("SAVE", (dialog, which) -> {
+        builder.setPositiveButton(R.string.save, (dialog, which) -> {
             String username = input.getText().toString();
             if (!username.isEmpty()) {
+                User user = User.fromDisk(this);
                 if (user != null && username.length() <= 20) {
```

Comment:
can we make a variable for the max amount of characters for username.

Reply:
Probably a good plan. :+1:

Reply:
We have one in `OnboardingUserActivity`, I just referred to that one.